File: src/models/evaluation.ts
import type { Skill } from "./skill";
import type { Subject } from "./subject";

export type Evaluation = Readonly<{
  name: string;
  id: string;
  teacher: string;
  coefficient: number;
  description: string;
  subject: Subject;
  /** @example ["Cycle 4"] */
  levels: Array<string>;
  skills: Array<Skill>;
  date: Date;
}>;

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/Evaluation.kt
package ink.literate.pawnote.models

import kotlinx.datetime.LocalDateTime

data class Evaluation(
    val name: String,
    val id: String,
    val teacher: String,
    val coefficient: Int,
    val description: String,
    val subject: Subject,
    /** Example: ["Cycle 4"] */
    val levels: List<String>,
    val skills: List<Skill>,
    val date: LocalDateTime
)
